File: python/supabase_connector.py
from supabase import create_client, Client
from typing import Dict, List, Any, Optional
import os

class SupabaseConnector:
    def __init__(self, supabase_url: str, supabase_key: str):
        self.supabase: Client = create_client(supabase_url, supabase_key)
    
    def insert(self, table_name: str, data: Dict[str, Any]) -> Dict[str, Any]:
        try:
            response = self.supabase.table(table_name).insert(data).execute()
            return {"success": True, "data": response.data}
        except Exception as e:
            return {"success": False, "error": str(e)}
    
    def select(self, table_name: str, query: Dict[str, Any] = None) -> Dict[str, Any]:
        try:
            request = self.supabase.table(table_name).select("*")
            
            if query and 'where' in query:
                for key, value in query['where'].items():
                    request = request.eq(key, value)
            
            response = request.execute()
            return {"success": True, "data": response.data}
        except Exception as e:
            return {"success": False, "error": str(e)}
    
    def update(self, table_name: str, query: Dict[str, Any], update_data: Dict[str, Any]) -> Dict[str, Any]:
        try:
            request = self.supabase.table(table_name).update(update_data)
            
            for key, value in query.items():
                request = request.eq(key, value)
            
            response = request.execute()
            return {"success": True, "data": response.data}
        except Exception as e:
            return {"success": False, "error": str(e)}
    
    def delete(self, table_name: str, query: Dict[str, Any]) -> Dict[str, Any]:
        try:
            request = self.supabase.table(table_name).delete()
            
            for key, value in query.items():
                request = request.eq(key, value)
            
            response = request.execute()
            return {"success": True, "data": response.data}
        except Exception as e:
            return {"success": False, "error": str(e)}

File: javascript/supabase-connector.js
const { createClient } = require('@supabase/supabase-js');

class SupabaseConnector {
  constructor(supabaseUrl, supabaseKey) {
    this.supabase = createClient(supabaseUrl, supabaseKey);
  }

  async insert(tableName, data) {
    try {
      const { data: result, error } = await this.supabase
        .from(tableName)
        .insert(data)
        .select();

      if (error) throw error;
      return { success: true, data: result };
    } catch (error) {
      return { success: false, error: error.message };
    }
  }

  async select(tableName, query = {}) {
    try {
      let request = this.supabase.from(tableName).select('*');
      
      // Filtros básicos
      if (query.where) {
        Object.keys(query.where).forEach(key => {
          request = request.eq(key, query.where[key]);
        });
      }

      const { data, error } = await request;

      if (error) throw error;
      return { success: true, data };
    } catch (error) {
      return { success: false, error: error.message };
    }
  }

  async update(tableName, query, updateData) {
    try {
      let request = this.supabase.from(tableName).update(updateData);
      
      Object.keys(query).forEach(key => {
        request = request.eq(key, query[key]);
      });

      const { data, error } = await request.select();

      if (error) throw error;
      return { success: true, data };
    } catch (error) {
      return { success: false, error: error.message };
    }
  }

  async delete(tableName, query) {
    try {
      let request = this.supabase.from(tableName).delete();
      
      Object.keys(query).forEach(key => {
        request = request.eq(key, query[key]);
      });

      const { data, error } = await request.select();

      if (error) throw error;
      return { success: true, data };
    } catch (error) {
      return { success: false, error: error.message };
    }
  }
}

module.exports = SupabaseConnector;